niworkflows/utils/images.py

```diff
@@ -195,6 +194,13 @@ def resample_by_spacing(in_file, zooms, order=3, clip=True):
         new_card.dot(np.vstack((new_grid, np.ones((1, new_grid.shape[1])))))
     )
 
+    if smooth:
+        from scipy.ndimage import gaussian_filter
+        data = gaussian_filter(in_file.get_fdata(),
```

Comment:
why the difference of `in_file.get_fdata()` and `in_file.dataobj`?

Reply:
`gaussian_filter` wants float, and nibabel does it conveniently. Please note the `astype` in the next line

Reply:
You'll probably want to round if re-casting to int, otherwise you truncate 2.9 to 2.